src/router.rs

```diff
@@ -19,16 +30,48 @@ pub fn hc_http_gateway_router(configuration: Configuration) -> Router {
 
 #[cfg(test)]
 pub mod tests {
+    use crate::holochain::AppCall;
     use crate::{
         config::{AllowedFns, Configuration, ZomeFn},
         router::hc_http_gateway_router,
+        AdminConn, HcHttpGatewayResult,
     };
     use axum::{body::Body, http::Request, Router};
+    use futures::future::BoxFuture;
+    use holochain::prelude::ExternIO;
+    use holochain_client::{AppWebsocket, SerializedBytes};
+    use holochain_serialized_bytes::prelude::*;
+    use holochain_types::app::InstalledAppId;
     use http_body_util::BodyExt;
     use reqwest::StatusCode;
+    use serde::{Deserialize, Serialize};
     use std::collections::{HashMap, HashSet};
+    use std::sync::Arc;
     use tower::ServiceExt;
 
+    #[derive(Debug, Serialize, Deserialize, SerializedBytes)]
+    pub struct TestZomeResponse {
+        hello: String,
+    }
+
+    #[derive(Debug)]
+    struct TestAppPool;
+
+    impl AppCall for TestAppPool {
```

Comment:
All these test types should probably be put somewhere a bit more obviously central, but this is what mocking the app pool could look like